source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,195 @@
+import React from 'dom-chef';
+import * as pageDetect from 'github-url-detection';
+import ChevronLeftIcon from 'octicon/chevron-left.svg';
+import ChevronRightIcon from 'octicon/chevron-right.svg';
+import select from 'select-dom';
+import cache from 'webext-storage-cache';
+
+import features from '.';
+import {getRepoURL} from '../github-helpers';
+import * as api from '../github-helpers/api';
+import SearchQuery from '../github-helpers/search-query';
+
+async function init() {
+	const previousButton = getButton('Navigate to previous Conversation', ChevronLeftIcon);
+	const nextButton = getButton('Navigate to next Conversation', ChevronRightIcon);
+
+	select('.gh-header-meta')?.append(
+		<div className="BtnGroup ml-2">
+			{previousButton}
+			{nextButton}
+		</div>,
+	);
+
+	const {list, listQuery} = await getConversationList();
+	const conversation = list.find(item => item.number === getConversationNumber());
+
+	setButtonHref(
+		previousButton,
+		listQuery,
+		list.find(item => item.cursor === alterCursor(conversation!.cursor, -1))
+	);
+
+	setButtonHref(
+		nextButton,
+		listQuery,
+		list.find(item => item.cursor === alterCursor(conversation!.cursor, +1))
+	);
+}
+
+function setButtonHref(button: HTMLAnchorElement, listQuery: ReturnType<typeof getListQuery>, conversation?: Conversation) {
+	if (conversation) {
+		const url = new URL(conversation.url);
+		url.searchParams.set('q', listQuery.query);
+		url.searchParams.set('page', listQuery.page.toString(10));
+		button.setAttribute('aria-disabled', 'false');
+		button.href = url.href;
+	}
+}
+
+const getButton = (ariaLabel: string, Icon: () => JSX.Element): HTMLAnchorElement => (
+	<a
+		aria-disabled="true"
+		className="btn btn-lg BtnGroup-item"
+		data-hotkey="n"
+		aria-label={ariaLabel}
+	>
+		<Icon/>
+	</a>
+) as any;
+
+// Current default number of items in a conversation list
+const ITEMS_PER_PAGE = 25;
+
+type Conversation = {
+	cursor: string;
+	number: number;
+	url: string;
+};
+
+const fetchConversationList = cache.function(
+	async (listQuery: ReturnType<typeof getListQuery>): Promise<Conversation[]> => {
+		const searchQuery = getConversationSearchQuery(listQuery);
+
+		return api.v4(searchQuery).then(({search}) => search.edges.map((edge: any) => ({
+			...edge.node,
+			cursor: edge.cursor
```

Comment:
From what I understand, the cursor is just a "stringified" number. Let's just store the number and only convert it the string when querying the API.

```suggestion
			cursor: getOffset(edge.cursor)
```

This makes operations easier

Reply:
Makes sense, so we can get rid of all cursor functions and do basic maths.
I'll take a look tommorow.